packages/web-config-server/src/export/excelFormatters/formatMatrixDataForExcel.js

```diff
@@ -124,12 +127,22 @@ const convertAoaToAoo = data => {
 
 const isEmpty = data => data === undefined || data === null;
 
-const addValueOrEmpty = value => {
+const addValueOrEmpty = (value, valueType) => {
   if (isEmpty(value)) return '';
 
   if (typeof value === 'object') {
+    if (valueType === VALUE_TYPES.NUMBER_AND_PERCENTAGE) {
+      const { value: val, metadata } = value;
+      return numberAndPercentage(val, metadata);
+    }
     return isEmpty(value.value) ? '' : value.value;
   }
 
   return value;
-}
+};
+
+const numberAndPercentage = (value, { numerator, denominator }) => {
+  if (isNaN(value)) return value;
+  const percentage = (numerator / denominator) * 100;
+  return `${value} (${percentage === 0 ? 0 : percentage.toFixed(1)}%)`;
+};
```

Comment:
I think it's better if we move this method to @tupaia/utils just in case we want to re use it.

Ideally I think we want to move all of the formatters in `formatters.js` from @tupaia/web-frontend to @tupaia/utils so we can reuse them in both web-frontend and web-config-server. But because of this issue (https://github.com/beyondessential/tupaia-backlog/issues/1798), web-frontend cannot use any internal dependency.

This also looks a bit simpler than the version in formatter.js as well...?

Reply:
Thanks Chris for sharing the latest news!
 
I'm not pretty familar about `formatters.js` on frontend but I guess it is for formatting the value before visualizing. If so I guess the best practice might be sending the value to frontend, after formating it instead of formating them on frontend? That's the thing in my head when looking at `formatters.js`. Unless there is any special case or other usage on front end.

Reply:
I think that would require a bit of refactor since we work out which formatter that we want to use based on `valueType` config in `viewJson`, so it should only apply in the front end

Reply:
Yes you are right, here we also use `valueType` in `viewJson`. Now we send `value: 0, { numerator, denominator}` then we calculate them to `0 (12%)` in front end. I just think that if there is a better solution to send `0 (12%)` directly to front end. 

And yeah if we do so we should do rafactor and a huge testing... In conclusion I'm happy to move that to `utils` 😄